uint32 bus_number
uint32[] port_numbers